### ISSUE TITLE
[rllib]How is the learning rate updated?

### ISSUE THREAD

**Issue by utkarshp** [ray-project/ray#10423]
I want to control the learning rate for the DQN algorithm, and implement something like torch.optim.lr_scheduler.ReduceLROnPlateau (but slightly more complex). I implemented a mixin for my policy and an execution plan for my trainer (based on DQNTorchPolicy and DQNTrainer respectively) to update the LR the way I want. However, I don't understand how to pass on that LR to the optimizer. I suppose I could just set policy._optimizer[0] to a new optimizer with the new learning rate, but I figure there must be a nicer way since RLlib already does have a LearningRateSchedule. So, I am trying to understand how that works.

When I set "lr_schedule" in the config, I see from stats that the learning rate is being updated. Through the code, I also see that the variable policy.cur_lr is being updated through the LearningRateSchedule mixin. However, I don't understand how this new cur_lr is passed on to the optimizer. I see optimizer() method in the same mixin, but that is never called. There is an adam_optimizer method in dqn_torch_policy, that is only called at initialization. I don't see any other place where this could be happening. Can someone please help?

Also, if what I described above is not the best way to do what I want to do, please let me know that too.

I am using Python version 3.8.5, Ray version 0.8.7, PyTorch version 1.4.0 on Debian GNU/Linux 10 (buster).

**Comment by Manuscrit** [comment]
I think that there may be an issue here. I have a similar problem.
With `DQNTorchPolicy`, the LR is never updated (cur_lr is updated but not used to update the optimizer).

The function `LearningRateSchedule.optimizer` is never called.
What is called is 
```
        @override(TorchPolicy)
        def optimizer(self):
            if optimizer_fn:
                optimizers = optimizer_fn(self, self.config)
            else:
                optimizers = TorchPolicy.optimizer(self)
            optimizers = force_list(optimizers)
            if hasattr(self, "exploration"):
                optimizers = self.exploration.get_exploration_optimizer(
                    optimizers)
            return optimizers
```
from torch_policy_template. The method is not overridden.

There was a possibly related issue: [https://github.com/ray-project/ray/pull/6101](https://github.com/ray-project/ray/pull/6101)
Where TorchPolicy needed to be changed to base but it doesn't look like this is going to solve this.
